File: test/day1_test.exs
defmodule Day1Test do
  use ExUnit.Case
  require Logger

  @example File.read!("priv/day1/example.txt") |> String.split("\n")
  @input File.read!("priv/day1/input.txt") |> String.split("\n")

  test "part 1" do
    assert Day1.part1(@example) == 24000
    Logger.info("Day 1, Part 1: #{Day1.part1(@input)}")
  end

  test "part 2" do
    assert Day1.part2(@example) == 45000
    Logger.info("Day 1, Part 2: #{Day1.part2(@input)}")
  end
end

File: kotlin/aoc/src/test/kotlin/y2024/Day1Test.kt
package y2024

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import utils.readFile

class Day1Test {
    private val day = Day1()

    @Test
    fun part1() {
        assertEquals(day.part1(readFile("/y2024/day1_sample.txt")), 11)
        assertEquals(day.part1(readFile("/y2024/day1.txt")), 1765812)
    }

    @Test
    fun part2() {
        assertEquals(day.part2(readFile("/y2024/day1_sample.txt")), 31)
        assertEquals(day.part2(readFile("/y2024/day1.txt")), 20520794)
    }
}
